src/filtertransdialog.cpp

```diff
@@ -1244,9 +1246,19 @@ bool mmFilterTransactionsDialog::mmIsRecordMatches(const Model_Checking::Data &t
     else if (mmIsNumberChecked() && (mmGetNumber().empty() ? !tran.TRANSACTIONNUMBER.empty()
         : tran.TRANSACTIONNUMBER.empty() || !tran.TRANSACTIONNUMBER.Lower().Matches(mmGetNumber().Lower())))
         ok = false;
-    else if (mmIsNotesChecked() && (mmGetNotes().empty() ? !tran.NOTES.empty()
-        : tran.NOTES.empty() || !tran.NOTES.Lower().Matches(mmGetNotes().Lower())))
-        ok = false;
+    else if (mmIsNotesChecked()) {
```

Comment:
If mmIsNotesChecked() = TRUE then the checking against the remainder checkboxes will not be processed. That is why the convoluted if statement was there I suspect.

Reply:
Good catch. I moved this to the end of the checks so nothing else is left to process.